router/core/factoryresolver.go

```diff
@@ -28,15 +28,23 @@ import (
 )
 
 type Loader struct {
+	ctx      context.Context
 	resolver FactoryResolver
 	// includeInfo controls whether additional information like type usage and field usage is included in the plan de
-	includeInfo bool
+	includeInfo               bool
+	logger                    *zap.Logger
+	addPubSubProviderCallback func(provider datasource.PubSubProvider)
+}
+
+type InstanceData struct {
```

Comment:
make struct private or expose fields publicly if used outside of this package, if just used in core, it can be all private